src/Extensions/ElementalAreasExtension.php

```diff
@@ -278,17 +280,21 @@ public function supportsElemental()
      */
     public function ensureElementalAreasExist($elementalAreaRelations)
     {
-        foreach ($elementalAreaRelations as $eaRelationship) {
-            $areaID = $eaRelationship . 'ID';
-
-            if (!$this->owner->$areaID) {
-                $area = ElementalArea::create();
-                $area->OwnerClassName = get_class($this->owner);
-                $area->write();
-                $this->owner->$areaID = $area->ID;
+        $owner = $this->getOwner();
+        // Only create the elemental area on draft stage
+        if (Versioned::get_reading_mode() === 'Stage.' . Versioned::LIVE) {
+            return $owner;
+        }
+        foreach ($elementalAreaRelations as $relation) {
+            if ($owner->$relation()->exists()) {
```

Comment:
1. Why is a new loop being added that wasn't there before?
2. Why are we changing to potentially fetching the record from the DB instead of using the `<relation>ID` column
3. Why are we calling `exists()` that wasn't called before?

None of these seem related to the cause of the bug as described in the PR description.